### DISCUSSION TITLE
Updating gradient offset has no effect in WASM

### DISCUSSION THREAD

**Discussion by andrewkdci** [unoplatform/uno#13906] [Q&A] [answered]
I'm not sure if this should be a bug report or if I'm just doing something wrong. So for now I'll start with this as a Q&A discussion.

In our application, we have a thermometer icon. We illustrate the temperature level by filling the icon with a LinearGradientBrush and then changing the Offset of the GradientStops. This works fine in WinUI, but changes to the gradient offset don't have any effect in WASM.

Below is an extremely simple app to demonstrate this:

1. Create a new Uno project titled "UnoGradientStop" with .NET 7 and targeting WinUI and WebAssembly (not hosted).
2. Update the MainPage.xaml and MainPage.xaml.cs with the following:

```
<Page
    x:Class="UnoGradientStop.MainPage"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:local="using:UnoGradientStop"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    mc:Ignorable="d"    
    Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">

    <Grid>
        <Border Height="50" Width="100"
                BorderBrush="Yellow"
                BorderThickness="2">
            <Border.Background>
                <LinearGradientBrush StartPoint="0,1" EndPoint="1,1">
                    <GradientStop x:Name="Stop1" Color="Transparent" Offset="0.5"/>
                    <GradientStop x:Name="Stop2" Color="Red" Offset="0.5"/>
                    <GradientStop x:Name="Stop3" Color="Red" Offset="1"/>
                </LinearGradientBrush>
            </Border.Background>
        </Border>
    </Grid>
</Page>
```

```
using Microsoft.UI.Xaml.Controls;
using System;
using System.Timers;

namespace UnoGradientStop
{
    public sealed partial class MainPage : Page
    {
        public MainPage()
        {
            this.InitializeComponent();
            var timer = new Timer(TimeSpan.FromMilliseconds(500));
            timer.Elapsed += Timer_Elapsed;
            timer.Start();
        }

        private double _stop = 0.5;

        private void Timer_Elapsed(object sender, ElapsedEventArgs e)
        {
            _stop += 0.1;
            if (_stop > 1)
                _stop = 0.3;

            DispatcherQueue.TryEnqueue(() =>
            {
                Stop1.Offset = _stop;
                Stop2.Offset = _stop;
            });
        }
    }
}
```

When run in WinUI, you see the gradient change every half second, as show in the .gif.

![UnoGradientStop](https://github.com/unoplatform/uno/assets/130710084/e7d11c6a-2058-4cdc-b13f-3e6220f5e53e)

 In WASM, the gradient does not change. It just stays static at its initial value.

Any idea on whether this is a limitation of Uno WASM (i.e. should it be a bug or feature request?), or am I just doing something wrong?

Thanks!

**Comment by jeromelaban** [comment, marked as answer]
Thanks for the report. [This is implemented](https://github.com/unoplatform/uno/blob/6afedadc731f8fd6efe44d27470e39bd7711ee54/src/Uno.UI/UI/Xaml/Media/GradientStop.cs#L56) in Uno 5.0 preview builds, take a look and let us know?

**Reply by andrewkdci** [reply]
Yep, it works in 5.0.0-dev.3407

Thanks!
